File: cci17.04-missing-number-lcci/cpp/Solution.cpp
class Solution {
public:
    int missingNumber(vector<int>& nums) {
        int n = nums.size();
        int ret = 0;
        for (int i = 0; i <= n; ++i) {
            ret ^= i;
        }
        for (int i = 0; i < n; ++i) {
            ret ^= nums[i];
        }
        return ret;
    }
};

File: cci17.04-missing-number-lcci/javascript/Solution.js
/**
 * @param {number[]} nums
 * @return {number}
 */
var missingNumber = function(nums) {
    let n = nums.length
    let ret = 0
    for (let i = 0; i <= n; ++i) {
        ret ^= i
    }
    for (let i = 0; i < n; ++i) {
        ret ^= nums[i]
    }
    return ret
};
